enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,320 @@
+--gzip \
+--file=upgrade-4.12.10-x86_64.tar.gz
+```
+
+That will pull approximately 180 images and requires 32 GiB of space for the images and another 16
+GiB for the tar file, so approximately 48 GiB in total.
+
+The engineer will write this tar file to some kind of media and hand it over to the technicians.
+
+#### Procedure to perform the upgrade
+
+The technician will receive the media containing the `upgrade-4.12.10-x86_64.tar.gz` file.
+
+The technician will verify that each node of the cluster has at least 32 GiB of space available in
+the `/var` filesystem, or 48 GiB if the tar file needs to be copied to the node in advance (if it
+isn't possible to plug and mount an USB stick, for example). This is crucial because if the node
+runs out of space performance will degrade and the node may stop functioning correctly.
+
+The technician will connect to the nodes of the cluster via SSH, create a `/var/lib/upgrade`
+directory in each node, and extract the upgrade package there:
+
+```
+# ssh core@node
+
+# sudo -i
+
+# mkdir /var/lib/upgrade
+
+# cd /var/lib/upgrade
+
+# tar \
+--extract \
+--xattrs \
+--xattrs-include="security.*" \
+--gzip \
+--file=/media/upgrade-4.12.10-x86_64.tar.gz
+```
+
+Note that it is very important to preserve capabilities and SELinux contexts in these files, so this
+needs to be done as root.
+
+The technician will change the configuration of CRI-O to use `/var/lib/upgrade` as an additional
+image store and to pin the release images. Inside `/etc/containers/storage.conf`:
+
+```toml
+additionalimagestores = [
+  /var/lib/upgrade/4.12.10/images
+]
+```
+
+Inside `/etc/crio/crio.conf.d/pin-upgrade.conf`:
+
+```toml
+pinned_images=[
+  quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:*
+]
+```
+
+The technician will then trigger the upgrade, adding the following to the cluster version object:
+
+```
+# oc edit clusterversion
+```
+
+```yaml
+desiredUpdate:
+  image: quay.io/openshift-release-dev/ocp-release@sha256:0398462b1c54758fe83619022a98bbe65f6deed71663b6665224d3ba36e43f03
+  version: 4.12.10
+```
+
+### Risks and Mitigations
+
+The proposed solution will require space to store all the release images in all the nodes of the
+cluster, approximately 48 GiB in the worst case. It will be necessary to calculate those space
+requirements in advance and ensure (via documentation or tooling) that the upgrade isn't attempted
+if the requirements aren't met.
+
+### Drawbacks
+
+The procedure to perform the upgrade is complicated and needs to be repeated in all the nodes of the
+cluster. Should we provide a tool automating that?
+
+## Design Details
+
+### Open Questions
+
+None.
+
+### Test Plan
+
+We should have at least tests that verifies that the upgrade can be performed in a fully
+disconnected environment, both for a single node cluster and a cluster with multiple nodes. These
+tests should gate the OCP release.
+
+It is desirable to have another test that scans the OCP components looking for use of the `Always`
```

Comment:
We should consider adding admission in CI for this.

Reply:
Done-